Fix random_double_fill function signature. https://github.com/numpy/numpy/issues/19312
This patch is no longer required for numpy 1.21.1

diff --git a/numpy/random/bit_generator.c b/numpy/random/bit_generator.c
index f53b246..7878b34 100644
--- a/numpy/random/bit_generator.c
+++ b/numpy/random/bit_generator.c
@@ -1163,15 +1163,15 @@ typedef enum __pyx_t_5numpy_6random_6common_ConstraintType __pyx_t_5numpy_6rando
 /* "common.pxd":53
  *     cdef void PyArray_free_aligned(void *p)
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  */
-typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
+typedef void (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
 
 /* "common.pxd":54
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
@@ -1179,7 +1179,7 @@ typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *,
 typedef double (*__pyx_t_5numpy_6random_6common_random_double_0)(void *);
 
 /* "common.pxd":55
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
diff --git a/numpy/random/bounded_integers.c b/numpy/random/bounded_integers.c
index 141d711..f126987 100644
--- a/numpy/random/bounded_integers.c
+++ b/numpy/random/bounded_integers.c
@@ -1142,15 +1142,15 @@ typedef enum __pyx_t_5numpy_6random_6common_ConstraintType __pyx_t_5numpy_6rando
 /* "common.pxd":53
  *     cdef void PyArray_free_aligned(void *p)
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  */
-typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
+typedef void (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
 
 /* "common.pxd":54
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
@@ -1158,7 +1158,7 @@ typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *,
 typedef double (*__pyx_t_5numpy_6random_6common_random_double_0)(void *);
 
 /* "common.pxd":55
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
diff --git a/numpy/random/common.c b/numpy/random/common.c
index 8ce7ebc..2d3dc15 100644
--- a/numpy/random/common.c
+++ b/numpy/random/common.c
@@ -1135,11 +1135,11 @@ typedef enum __pyx_t_5numpy_6random_6common_ConstraintType __pyx_t_5numpy_6rando
 /* "numpy/random/common.pxd":53
  *     cdef void PyArray_free_aligned(void *p)
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  */
-typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
+typedef void (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
 
 /* "numpy/random/common.pxd":54
  * 
diff --git a/numpy/random/common.pxd b/numpy/random/common.pxd
index ac0a94b..4c2d651 100644
--- a/numpy/random/common.pxd
+++ b/numpy/random/common.pxd
@@ -50,7 +50,7 @@ cdef extern from "src/aligned_malloc/aligned_malloc.h":
     cdef void *PyArray_calloc_aligned(size_t n, size_t s)
     cdef void PyArray_free_aligned(void *p)
 
-ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
 ctypedef double (*random_double_0)(void *state) nogil
 ctypedef double (*random_double_1)(void *state, double a) nogil
 ctypedef double (*random_double_2)(void *state, double a, double b) nogil
diff --git a/numpy/random/generator.c b/numpy/random/generator.c
index 585659a..4ef255a 100644
--- a/numpy/random/generator.c
+++ b/numpy/random/generator.c
@@ -1249,15 +1249,15 @@ typedef enum __pyx_t_5numpy_6random_6common_ConstraintType __pyx_t_5numpy_6rando
 /* "common.pxd":53
  *     cdef void PyArray_free_aligned(void *p)
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  */
-typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
+typedef void (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
 
 /* "common.pxd":54
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
@@ -1265,7 +1265,7 @@ typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *,
 typedef double (*__pyx_t_5numpy_6random_6common_random_double_0)(void *);
 
 /* "common.pxd":55
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
diff --git a/numpy/random/mt19937.c b/numpy/random/mt19937.c
index 17e9eab..9afacb9 100644
--- a/numpy/random/mt19937.c
+++ b/numpy/random/mt19937.c
@@ -1127,15 +1127,15 @@ typedef enum __pyx_t_5numpy_6random_6common_ConstraintType __pyx_t_5numpy_6rando
 /* "common.pxd":53
  *     cdef void PyArray_free_aligned(void *p)
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  */
-typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
+typedef void (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
 
 /* "common.pxd":54
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
@@ -1143,7 +1143,7 @@ typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *,
 typedef double (*__pyx_t_5numpy_6random_6common_random_double_0)(void *);
 
 /* "common.pxd":55
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
diff --git a/numpy/random/mtrand.c b/numpy/random/mtrand.c
index 36995bf..1824e51 100644
--- a/numpy/random/mtrand.c
+++ b/numpy/random/mtrand.c
@@ -1147,15 +1147,15 @@ typedef enum __pyx_t_5numpy_6random_6common_ConstraintType __pyx_t_5numpy_6rando
 /* "common.pxd":53
  *     cdef void PyArray_free_aligned(void *p)
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  */
-typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
+typedef void (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
 
 /* "common.pxd":54
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
@@ -1163,7 +1163,7 @@ typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *,
 typedef double (*__pyx_t_5numpy_6random_6common_random_double_0)(void *);
 
 /* "common.pxd":55
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
diff --git a/numpy/random/pcg64.c b/numpy/random/pcg64.c
index 2887b9f..cb948b9 100644
--- a/numpy/random/pcg64.c
+++ b/numpy/random/pcg64.c
@@ -1127,15 +1127,15 @@ typedef enum __pyx_t_5numpy_6random_6common_ConstraintType __pyx_t_5numpy_6rando
 /* "common.pxd":53
  *     cdef void PyArray_free_aligned(void *p)
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  */
-typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
+typedef void (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
 
 /* "common.pxd":54
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
@@ -1143,7 +1143,7 @@ typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *,
 typedef double (*__pyx_t_5numpy_6random_6common_random_double_0)(void *);
 
 /* "common.pxd":55
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
diff --git a/numpy/random/philox.c b/numpy/random/philox.c
index 4ce548f..862a390 100644
--- a/numpy/random/philox.c
+++ b/numpy/random/philox.c
@@ -1127,15 +1127,15 @@ typedef enum __pyx_t_5numpy_6random_6common_ConstraintType __pyx_t_5numpy_6rando
 /* "common.pxd":53
  *     cdef void PyArray_free_aligned(void *p)
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  */
-typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
+typedef void (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
 
 /* "common.pxd":54
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
@@ -1143,7 +1143,7 @@ typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *,
 typedef double (*__pyx_t_5numpy_6random_6common_random_double_0)(void *);
 
 /* "common.pxd":55
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
diff --git a/numpy/random/sfc64.c b/numpy/random/sfc64.c
index a235d52..445c308 100644
--- a/numpy/random/sfc64.c
+++ b/numpy/random/sfc64.c
@@ -1127,15 +1127,15 @@ typedef enum __pyx_t_5numpy_6random_6common_ConstraintType __pyx_t_5numpy_6rando
 /* "common.pxd":53
  *     cdef void PyArray_free_aligned(void *p)
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  */
-typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
+typedef void (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *, npy_intp, double *);
 
 /* "common.pxd":54
  * 
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_1)(void *state, double a) nogil
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
@@ -1143,7 +1143,7 @@ typedef double (*__pyx_t_5numpy_6random_6common_random_double_fill)(bitgen_t *,
 typedef double (*__pyx_t_5numpy_6random_6common_random_double_0)(void *);
 
 /* "common.pxd":55
- * ctypedef double (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
+ * ctypedef void (*random_double_fill)(bitgen_t *state, np.npy_intp count, double* out) nogil
  * ctypedef double (*random_double_0)(void *state) nogil
  * ctypedef double (*random_double_1)(void *state, double a) nogil             # <<<<<<<<<<<<<<
  * ctypedef double (*random_double_2)(void *state, double a, double b) nogil
